src/Symfony/Component/Console/Tests/ColorTest.php

```diff
@@ -48,16 +48,31 @@ public function testDegradedTrueColors()
     {
         $colorterm = getenv('COLORTERM');
         $term = getenv('TERM');
-        putenv('COLORTERM=');
-        putenv('TERM=');
 
         try {
             // To Ansi4
+            putenv('COLORTERM=');
+            putenv('TERM=');
+
             $color = new Color('#f00', '#ff0');
             $this->assertSame("\033[31;43m \033[39;49m", $color->apply(' '));
 
             $color = new Color('#c0392b', '#f1c40f');
             $this->assertSame("\033[31;43m \033[39;49m", $color->apply(' '));
+
+            // To Ansi 8, parsed from $TERM
+            putenv('COLORTERM=');
```

Comment:
Please do that as a separate test. We should not mix degrading to 4 bits or to 8 bits in the same test. That makes the test a lot less readable.

Reply:
as it is a public API, it should indeed have its own tests